workflow/models/time_entry.py

```diff
@@ -69,8 +69,8 @@ class Meta:
         ordering = ["created_at"]  # Default ordering by creation time
 
     def save(self, *args, **kwargs):
-        # Ensure minutes and hours stay consistent
-        if self.items is not None and self.minutes_per_item is not None:
+        # Only calculate hours from items/minutes if hours isn't explicitly set
```

Comment:
Why?  

Again.  Do not rewrite code in comments

Reply:
I updated all the comments you mentioned to improve code readability while also explaining my intention in the sections where I thought it made sense to do this.

Reply:
Also, in the main changes I made and in the new functions / functions I changed a lot I decided to add more detailed documentation (especially in the JS functions), but still explaining the thought process / business logic / intention of the code.